tests/Kernel/Patterns/MediaContentPatternRenderingTest.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_theme\Kernel\Patterns;
+
+use Drupal\Tests\oe_theme\Kernel\AbstractKernelTestBase;
+
+/**
+ * Test media content pattern rendering.
+ */
+class MediaContentPatternRenderingTest extends AbstractKernelTestBase {
+
+  /**
+   * Test that media patterns are correctly rendered when passing an URL object.
+   *
+   * @param array $media
+   *   Media data for the pattern.
+   * @param array $assertions
+   *   Test assertions.
+   *
+   * @throws \Exception
+   *
+   * @dataProvider dataProvider
+   */
+  public function testMediaPattern(array $media, array $assertions) {
+    $pattern = [
+      '#type' => 'pattern',
+      '#id' => 'media_container',
+      '#fields' => [
+        'description' => $media['description'],
+        'media' => $media['media'],
```

Comment:
isn't this expecting a MediaValueObject object in the pattern definition?
we could create a media object and then pass it here
```suggestion
        'media' => MediaValueObject::fromMediaObject($media_object),
```

see https://github.com/openeuropa/oe_theme/blob/2.x/tests/Kernel/Patterns/FilePatternRenderingTest.php#L55

lets discuss